File: android-core/src/androidTest/java/com/mparticle/internal/MPUtilityTest.java
package com.mparticle.internal;

import android.content.Context;
import android.telephony.TelephonyManager;

import com.mparticle.testutils.BaseCleanInstallEachTest;

import org.junit.Test;

import java.util.HashMap;
import java.util.Hashtable;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.TreeMap;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class MPUtilityTest extends BaseCleanInstallEachTest {

    @Test
    public void testInstantAppDetectionTest() {
        assertFalse(MPUtility.isInstantApp(mContext));
    }

    @Test
    public void testNullMapKey() throws Exception {
        Map map = new HashMap();
        map.put("key1", "val1");
        map.put("key2", "val2");
        assertFalse(MPUtility.containsNullKey(map));
        map.put(null, "val3");
        assertTrue(MPUtility.containsNullKey(map));

        map = new Hashtable();
        map.put("key1", "val1");
        map.put("key2", "val2");
        assertFalse(MPUtility.containsNullKey(map));

        map = new TreeMap(map);
        assertFalse(MPUtility.containsNullKey(map));

        map = new LinkedHashMap(map);
        assertFalse(MPUtility.containsNullKey(map));
    }

    @Test
    public void testGetInstrumentedNetworkType() throws Exception {
        TelephonyManager manager = (TelephonyManager) mContext.getSystemService(Context.TELEPHONY_SERVICE);
        Integer result = MPUtility.getNetworkType(mContext, manager);
        assertNull(result);
    }
}

File: android-core/src/androidTest/java/com/mparticle/internal/MPUtilityTest.kt
package com.mparticle.internal

import android.telephony.TelephonyManager
import com.mparticle.testing.BaseTest
import com.mparticle.testing.context
class MPUtilityTest : BaseTest() {
    @org.junit.Test
    fun testInstantAppDetectionTest() {
        org.junit.Assert.assertFalse(MPUtility.isInstantApp(context))
    }

    @org.junit.Test
    @Throws(java.lang.Exception::class)
    fun testNullMapKey() {
        var map = mutableMapOf<String?, String?>(
            "key1" to "val1",
            "key2" to "val2"
        )
        org.junit.Assert.assertFalse(MPUtility.containsNullKey(map))
        map[null] = "val3"
        org.junit.Assert.assertTrue(MPUtility.containsNullKey(map))
        map = mutableMapOf()
        map["key1"] = "val1"
        map["key2"] = "val2"
        org.junit.Assert.assertFalse(MPUtility.containsNullKey(map))
        map = java.util.TreeMap(map)
        org.junit.Assert.assertFalse(MPUtility.containsNullKey(map))
        map = java.util.LinkedHashMap(map)
        org.junit.Assert.assertFalse(MPUtility.containsNullKey(map))
    }

    @org.junit.Test
    @Throws(java.lang.Exception::class)
    fun testGetInstrumentedNetworkType() {
        val manager: TelephonyManager =
            context.getSystemService(android.content.Context.TELEPHONY_SERVICE) as TelephonyManager
        val result = MPUtility.getNetworkType(context, manager)
        org.junit.Assert.assertNull(result)
    }
}
